app/Jobs/Notifications/UserAchievementUnlock.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Jobs\Notifications;
+
+use App\Models\Achievement;
+use App\Models\User;
+
+class UserAchievementUnlock extends BroadcastNotificationBase
+{
+    public function __construct(Achievement $object, User $source)
+    {
+        parent::__construct($object, $source);
```

Comment:
the old one has source reset to new user so notification server will still forward it (there's user id check in `user-connection.ts`). Though considering notifications are now sent to per-user channel maybe the check is not needed anymore in the notification server side and this can stay the same user?